spec/linkedlist.js

```diff
@@ -0,0 +1,195 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import LinkedList from '../src/linkedlist'
+
+chai.use(chaiChange)
+
+const generateTestList = () => {
+  const list = new LinkedList()
+
+  list.insert('first')
+  list.insert('middle')
+  list.insert('last')
+
+  return list
+}
+
+
+describe('LinkedList', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(LinkedList).to.be.a('function')
+  })
+
+  describe('insertFirst', () => {
+
+    it('Inserts a node with the provided data to the head of the list', () => {
+      const aLinkedList = new LinkedList()
+
+      expect(() => aLinkedList.insertFirst('foo'))
+        .to.alter(() => aLinkedList.size(), { from: 0, to: 1 })
+    })
+
+    it('Inserts a node with the provided data to the head of the list', () => {
+      const aLinkedList = new LinkedList()
+
+      aLinkedList.insertFirst('bar')
+
+      expect(() => aLinkedList[0]
+        .to.equal('bar'))
+    })
+  })
+
+  describe('insert()', () => {
+    it('Inserts a node (with the provided data) to the tail of the list', () => {
+      const aLinkedList = generateTestList()
+
+      expect(() => aLinkedList.insert('junk'))
+        .to.alter(() => aLinkedList.size(), { from: 3, to: 4 })
+    })
+
+    it('Inserts a node (with the provided data) to the tail of the list', () => {
+      const aLinkedList = generateTestList()
+
+      aLinkedList.insert('junk')
+
+      expect(aLinkedList.tail.data)
+        .to.equal('junk')
+    })
+  })
+
+  describe('find()', () => {
+    it('Returns -1 if not found', () => {
+      const aLinkedList = generateTestList()
+
+      expect(aLinkedList.find('doesntexist'))
+        .to.equal(-1)
+    })
+
+    it('Returns the first node containing the provided data', () => {
```

Comment:
May want to include a test to verify that it is indeed the first node you receive if there are duplicate data values (e.g., what happens if there are two nodes containing `middle` in the `data` property?).